logstash-core/spec/logstash/queue_factory_spec.rb

```diff
@@ -72,9 +74,10 @@
   context "when `queue.type` is `memory`" do
     before do
       settings.set("queue.type", "memory")
+      settings.set("pipeline.batch.size", 1024)
```

Comment:
@jordansissel yea, because the size of the `ArrayBlockingQueue` is set as twice the product of this setting and the worker count (obviously there are many ways to size this, it just seemed like a reasonable starting point).